src/versionize_derive/src/fields/struct_field.rs

```diff
@@ -136,9 +138,11 @@ impl StructField {
 
                 quote! {
                     #field_ident: {
-                        let v: Vec<#array_type_token> = <Vec<#array_type_token> as Versionize>::deserialize(&mut reader, version_map, app_version)?;
-                        vec_to_arr_func!(transform_vec, #array_type_token, #array_len);
-                        transform_vec(v)
+                        let mut array = [#array_type_token::default() ; #array_len];
```

Comment:
I don't think we should rely on the assumption that array members always implement `Default` even though the alternatives are more verbose/complex :( One option coming to mind is to rely on macros somewhere (i.e. the primitives file) to implement `Versionize` for `[T; 0]` to `[T; 32]` (which seems to be standard practice when adding trait implementation for arrays, since Rust doesn't accept generics in terms of numerical values). I think the same applies to tuples, up to a certain cardinality.

Reply:
I am fine with the current limitation and the simple implementation as it handles any array size without having to generate impls for `Versionize for [T; 0] to [T; 32]`.  

I'll try someting when I implement tuples.

Reply:
Cool, maybe we can mention this in the issue you referenced before as well.

Reply:
#1728